### PR TITLE
Modify NodeSource shutdown

### PR DESCRIPTION
**Pull request by jrochas**
- remove callback from node source to node source policy shutdown, as it is suspected to be source of deadlock in the RM shutdown, instead the shutdown is asynchronously triggered from the node source and the node source active object is terminated just after.
- modify NodeSourcePolicy so that it does not require a stub reference at shutdown time
- remove the conditional trigger for NodeSource shutdown in the NodeSource#removeNode method. We could not identify a need for it to be triggered there.
- rename variables in RMCore#shutdown to better identify node source and node source name

### CODE REVIEW

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java [hunk 1]
```diff
@@ -765,6 +761,8 @@ protected void shutdownNodeSourceServices(Client initiator) {
 
         this.activePolicy.shutdown(initiator);
         this.infrastructureManager.internalShutDown();
+        this.rmcore.disconnect(Client.getId(PAActiveObject.getStubOnThis()));
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: maybe it's better to do this inside finishNodeSourceShutdown ?

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/7f1ead7149dd4da82d37a08cd6c024ca707283cf/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [NodeSource.java#L639](https://github.com/ow2-proactive/scheduling/blob/7f1ead7149dd4da82d37a08cd6c024ca707283cf/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L639): Remove this unused method parameter "initiator". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java (hunk 1 above)
Comment (jrochas) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: Yes it makes more sense. Applied.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/2726b3f2bb2073f5f4f015a70a0c1eafb7a0525e/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [NodeSource.java#L639](https://github.com/ow2-proactive/scheduling/blob/2726b3f2bb2073f5f4f015a70a0c1eafb7a0525e/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L639): Remove this unused method parameter "initiator". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/policy/NodeSourcePolicy.java [hunk 2]
```diff
@@ -122,9 +124,7 @@ public void reconfigure(Object... updatedPolicyParameters) throws Exception {
      * Shutdown the policy
      */
     public void shutdown(Client initiator) {
```
Comment (Mykhailenko) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/policy/NodeSourcePolicy.java]: you do not use any more argument

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/95749da76d96a6ac70b9378053276f00e41f9e0b/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [NodeSource.java#L639](https://github.com/ow2-proactive/scheduling/blob/95749da76d96a6ac70b9378053276f00e41f9e0b/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L639): Remove this unused method parameter "initiator". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by jrochas** [comment]
retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/51b73f528e8cc5f36fcfeca99031ca11e9648993/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [NodeSource.java#L639](https://github.com/ow2-proactive/scheduling/blob/51b73f528e8cc5f36fcfeca99031ca11e9648993/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L639): Remove this unused method parameter "initiator". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**